intranet/apps/events/tasks.py

```diff
@@ -0,0 +1,89 @@
+from datetime import date, datetime
+
+import requests
+from celery import shared_task
+from celery.utils.log import get_task_logger
+
+from django.utils import timezone
+
+from .models import Event
+
+logger = get_task_logger(__name__)
+
+
+@shared_task
+def pull_sports_schedules(month=None) -> None:
+
+    start_date = None
+    today = date.today()
+    if month is None:
+        start_date = today.replace(month=today.month + 1, day=1) if today.month < 12 else today.replace(month=1, day=1)
+    else:
+        start_date = date(year=today.year, month=month, day=1)
+
+    logger.info("Pulling sports schedules for the month")
+
+    r = requests.post(
+        "https://srv1-advancedview.rschooltoday.com/public/conference/downloadnow",
+        data={
+            "G5genie": "202",
+            "G5button": "13",
+            "ffpage": "advanced",
+            "G5MID": """052065101117117048054105068100043047108118057077075113047088099122077084047069105052098
+            112080115114075057121117110097100080066109083073066051114117112065043066081082050066067114098105081""",
+            "school_id": "9",
+            "category": "0",
+            "preview": "no",
+            "vw_activity": "0",
+            "vw_conference_events": "1",
+            "vw_non_conference_events": "1",
+            "vw_homeonly": "1",
+            "vw_awayonly": "1",
+            "vw_schoolonly": "1",
+            "vw_gender": "1",
+            "vw_type": "0",
+            "vw_level": "0",
+            "category_sel": "0",
+            "vw_opponent": "0",
+            "vw_location_detail": "0",
+            "opt_show_location": "1",
+            "opt_show_comments": "0",
+            "opt_show_bus_times": "0",
+            "opt_show_changes_cancellations": "0",
+            "vw_location": "0",
+            "vw_period": "month-yr",
+            "vw_month2": str(start_date),
+            "vw_monthCnt": "01",
+            "test": "test",
+            "multipleSchools": "0",
+            "sortType": "time",
+            "expandView": "0",
+            "listact": "0",
+            "urlPerl": "-www-northernregionva-org",
+            "ffVwLayout": "1",
+            "downloadtype": "csv",
+        },
+    )
+
+    games = [
```

Comment:
You may be better off using Python's `csv` library?

```python
import csv
import io

reader = csv.DictReader(r.content.decode("UTF-8").splitlines())

for line in reader:
    print(line)
```

Output:

```
{'Subject': 'Basketball:  Boys Varsity (Date & Time Changed from 01-20-21) ', 'Start Date': '02/03/2021', 'Start Time': '7:00pm', 'End Date': '02/03/2021', 'End Time': '', 'Description': ' Type: Game Opponent: George Marshall', 'Location': 'Thomas Jefferson Science & Technology High School'}
{'Subject': 'Basketball:  Girls Varsity (Rescheduled from 01-21-21) ', 'Start Date': '02/03/2021', 'Start Time': '7:00pm', 'End Date': '02/03/2021', 'End Time': '', 'Description': ' Type: Game Opponent: Falls Church', 'Location': 'Falls Church High School'}
{'Subject': 'Gymnastics:  Girls -  ', 'Start Date': '02/05/2021', 'Start Time': 'TBD', 'End Date': '02/05/2021', 'End Time': '', 'Description': ' Type: Region Finals Opponent: Woodgrove', 'Location': 'Woodgrove High School'}
{'Subject': 'Indoor Track:  Varsity  ', 'Start Date': '02/06/2021', 'Start Time': '12:00pm', 'End Date': '02/06/2021', 'End Time': '', 'Description': ' Type: District Tournament Opponent: Away vs. Multiple Schools', 'Location': 'Falls Church High School'}
{'Subject': 'Indoor Track:  Varsity  ', 'Start Date': '02/08/2021', 'Start Time': 'TBD', 'End Date': '02/08/2021', 'End Time': '', 'Description': ' Type: District Tournament Opponent: Away vs. Multiple Schools', 'Location': 'Falls Church High School'}
{'Subject': 'Basketball:  Girls Varsity  ', 'Start Date': '02/08/2021', 'Start Time': '5:30pm', 'End Date': '02/08/2021', 'End Time': '', 'Description': ' Type: Region 1st Round Opponent: Briar Woods', 'Location': 'Thomas Jefferson Science & Technology High School'}
{'Subject': 'Football:  Varsity  ', 'Start Date': '02/22/2021', 'Start Time': '7:00pm', 'End Date': '02/22/2021', 'End Time': '', 'Description': ' Type: Game Opponent: Lightridge', 'Location': 'Lightridge High School 41025 Collaboration Drive Aldie, VA 20105A'}
{'Subject': 'Cheerleading:  Varsity  ', 'Start Date': '02/22/2021', 'Start Time': '7:00pm', 'End Date': '02/22/2021', 'End Time': '', 'Description': ' Type: Game Opponent: Lightridge', 'Location': 'Lightridge High School 41025 Collaboration Drive Aldie, VA 20105A'}
```

Reply:
Is there a difference in performance?

Reply:
I don't think so, but it might be simpler to use (in my opinion, more straightforward than two list comprehensions and in case VHSL (or whatever SaaS service is used) changes the format or the order of the columns it will still work out).

Reply:
Aww list comprehensions make me feel powerful, but I suppose you are right. Changes made.